### ISSUE TITLE
Target for minimum percent of planning units

### ISSUE THREAD

**Issue by jflowernet** [prioritizr/prioritizr#163]
Just wanted to check: if you want to set a target for the minimum % of the entire landscape (i.e. planning units) that you want included in the solution, I think the best way to do that would be to have a dummy feature that has uniform value and set a target for that?

**Comment by jeffreyhanson** [comment]
Yeah that would be the best way to do it under the minimum set objective. If you're dealing with a budget-limited objective (e.g. max features), then you might need to play with the weights to do this.

**Comment by jflowernet** [comment]
Cool, thanks. 
Am I correct in saying that if, for example, we set a target for the dummy feature of 20%, and the other features we are prioritizing only get us to 15%, the extra 5% would be randomly chosen but constrained to minimize cost and boundary penalties?

**Comment by jeffreyhanson** [comment]
Yeah - that's correct. If you had equal cost data values and no extra penalties, then the extra planning units probably wouldn't be "truly random" though (e.g. Gurobi tends to select extra planning units that are next to each other in the ILP format in absence of additional information, so you can get "lines" appearing across the solutions).

**Comment by jflowernet** [comment]
Ahhh, that's useful info. I was wondering why I was getting solutions that were blocked together even without penalties in another run of prioritizr I was doing. It seems like it just 'fills up the bath', starting at one corner and choosing neighbouring planning units in lines going up.
Thanks Jeff!

**Comment by ricschuster** [comment]
Once I realized what's causing those strange patterns, I started to really like that behavior of the ILP solvers (Symphony does the same thing).

As to your question @jflowernet, this could be due to uniform cost and a feature not overlapping any other features. If prioritizr needs to fill up that feature and there is no useful information present to help distinguish different pixels, it will go ahead and fill things up for that feature. This can happen when feature data at a coarser resolution is down-sampled. 

Its really useful to see the unintended 'features' of the ILP solvers present themselves, as they do help (me) scrutinize the input data better than what we were able to do with Simulated Annealing solvers, which don't show those patterns quite as pronounced. What you do get with SA is very low selection frequencies though, which is a decent indicator of this type of problem as well.
